### PR TITLE
feat: add additional content for lottery published email

### PR DESCRIPTION
**Pull request by jaredcwhite**
This PR addresses #838

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

This updates the email that applicants receive when a lottery is published.

## How Can This Be Tested/Reviewed?

* Reseed the backend and start it up
* Log in as a admin user (either through the partner UI or directly in the swagger doc)
* In the swagger doc http://localhost:3100/api#/scriptRunner/addDuplicatesInformationToLotteryEmail run the script
* The new translation keys should be in the database in the `translations` table
* I'm not sure what's the easiest way to verify the email — it goes out to applicants when a listing is published

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [ ] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [x] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [x] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

**Comment by jaredcwhite** [comment]
I don't understand why the backend unit test is failing…it seems to be some problem with auth tokens, which I don't believe any of my code affected.

**Comment by emilyjablonski** [comment]
We can only run scripts once, so we will have to create a new script that runs with the strings, similar to how we have `addLotteryTranslations` and `addLotteryTranslationsCreateIfEmpty`. There's some extra documentation [here](https://www.notion.so/exygy/How-to-add-and-run-a-script-in-the-script-runner-abc7520d708344dd94d519423f65a066?pvs=4) and [this PR](https://github.com/bloom-housing/bloom/pull/4283) may be helpful as well.

**Comment by jaredcwhite** [comment]
Thanks @emilyjablonski I wasn't exactly sure what approach to use here. I'll give that a look and fix.

**Comment by jaredcwhite** [comment]
OK, I've backed out of updating translations in the old script, and created a new script endpoint. I'm not really sure how to test it though. A few other tests I looked through for the script runner seem to validate that the script was run, but mock all the DB access. I don't know how to validate I actually updated translations successfully without mangling the JSON somehow.

**Comment by YazeedLoonat** [comment]
Hey @jaredcwhite a couple things:
1. you're missing exposing your new service function in the api. You'd need to create that endpoint in the script runner controller where you can more or less copy how the other script runner endpoints live. That would all you to test locally but hitting the api targeting that new endpoint 

2. You should add unit tests for your new script runner service function. We do mock up the db reads/writes in our unit tests but thats mostly so we can isolate logic and strongly recommend test coverage for all script runner service functions. Since they are built to only run once, if it fails in prod or produces wonky results it would be a pain to then fix

**Comment by jaredcwhite** [comment]
Thanks @YazeedLoonat for the pointers! 🙏🏻  I'll look into all of that.

**Comment by jaredcwhite** [comment]
@YazeedLoonat OK, I should be good to go with those two points, although the actual verification of what the updated JSON looks like in the DB I had to do manually when testing locally. Happy to keep iterating if you have a suggestion.

@KrissDrawing I think you can start working on #837 either by going off of this branch, or creating a new one with a similar approach. Whichever seems easiest.

**Comment by mcgarrye** [comment]
Please add description and testing info to the PR template. Merge conflicts need addressing

**Comment by emilyjablonski** [comment]
From the PR linked above, we already had the `updateForLanguage` helper - I wonder if we should re-use that? There's some duplication going on with `updateLanguage`

File: api/src/services/script-runner.service.ts [hunk 1]
```diff
@@ -1105,6 +1105,86 @@ export class ScriptRunnerService {
...
+  async addDuplicatesInformationToLotteryEmail(
+    req: ExpressRequest,
+  ): Promise<SuccessDTO> {
+    const requestingUser = mapTo(User, req['user']);
+    await this.markScriptAsRunStart(
+      'add duplicates information to lottery email',
+      requestingUser,
+    );
+
+    const updateTranslation = async (
```
Comment (YazeedLoonat) [inline api/src/services/script-runner.service.ts:1123]: lets not stack helper functions in a function. This should be pulled out into a stand alone helper function

**Comment by jaredcwhite** [comment]
OK, this is ready for review again!

File: api/src/services/email.service.ts [hunk 2]
```diff
@@ -672,7 +672,8 @@ export class EmailService {
             appUrl: jurisdiction.publicUrl,
           },
           appUrl: jurisdiction.publicUrl,
-          // These two URLs are placeholders and must be updated per jurisdiction
+          // These three URLs are placeholders and must be updated per jurisdiction
```
Comment (ludtkemorgan) [inline api/src/services/email.service.ts:675]: Maybe the comment should be updated now that we have a specific url

**Review by ludtkemorgan — APPROVED** [review]
Just small comments. Also, this needs to be rebased with main, another script runner has been merged

File: api/src/services/script-runner.service.ts [hunk 3]
```diff
@@ -1044,7 +1031,7 @@ export class ScriptRunnerService {
   async addLotteryTranslations(req: ExpressRequest): Promise<SuccessDTO> {
     const requestingUser = mapTo(User, req['user']);
     await this.markScriptAsRunStart('add lottery translations', requestingUser);
-    this.addLotteryTranslationsHelper(req);
+    this.addLotteryTranslationsHelper();
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply api/src/services/script-runner.service.ts:1034]: The initial endpoint didn't account for the translations not already existing and didn't work, so we needed another that checked if they needed to be created. But the second existing fxn should do that on its own

File: api/src/services/email.service.ts (hunk 2 above)
Comment (jaredcwhite) [reply api/src/services/email.service.ts:675]: I'm not following. Do you mean the notificationsUrl and helpCenterUrl will be identical?

File: api/src/services/script-runner.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaredcwhite) [reply api/src/services/script-runner.service.ts:1034]: @emilyjablonski @ludtkemorgan So just so I'm clear, I don't need to make a change here?

[the remark this answers is not in the document]
Comment (emilyjablonski) [reply api/src/services/script-runner.service.ts]: It looks like the old `updateForLanguage` did it automatically and this new one has a parameter `createIfMissing` that will only do it if that's true - so we should pass in true

File: api/src/services/email.service.ts (hunk 2 above)
Comment (ludtkemorgan) [reply api/src/services/email.service.ts:674]: Oh sorry, I just meant that one of the three URLs with this change now has the non-placeholder URL so it's just the other two urls that need to be updated at a future date

File: api/src/services/script-runner.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ludtkemorgan) [reply api/src/services/script-runner.service.ts]: We should probably pass in `true` here just to have this feature be available in all languages. Hopefully this is the last time we have to use this though if we are able to tackle this ticket https://github.com/bloom-housing/bloom/issues/4452

**Pull request merged**